Yoast/wordpress-seo
# Merge Czech feature branch

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Adds Czech to the supported languages list and 3 eslint comments. It also modifies the path to the Czech researcher by using the locale `cs`.
* Lets Free users see an upsell notification that promotes Premium word form recognition. Reduce the limit of prominent words that are saved so that it matches the limit of other languages with function word support (languages with no function words get a higher limit to increase the chances of content words showing up in the prominent words).

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* This PR can be acceptance tested by following these steps:


Test a post in Czech in following environment:

- A normal post
- Taxonomy page

Steps to test:
- Confirm that the SEO and readability analysis load
- Set a keyphrase, add it to the text and see that it's recognized
- Change some readability metrics (e.g., add transition words) and confirm that the analysis updates accordingly
- Check that you are seeing a morphology upsell. (See [this PR](https://github.com/Yoast/wordpress-seo/pull/15322) for what the upsell looks like).


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-618
